src/facade/dragonfly_connection.cc

```diff
@@ -612,7 +624,12 @@ void Connection::RegisterBreakHook(BreakerCb breaker_cb) {
 }
 
 std::pair<std::string, std::string> Connection::GetClientInfoBeforeAfterTid() const {
-  CHECK(service_ && socket_);
+  if (!socket_) {
+    LOG(DFATAL) << "unexpected null socket_ "
+                << " phase " << unsigned(phase_) << ", is_http: " << unsigned(is_http_);
+    return {};
+  }
```

Comment:
So this can become a CHECK again?

Reply:
It can, but I do not see a point. The code will work fine if we just return an empty pair. #avoidredundantchecks

Reply:
But to some extent this is a declaration that you're still not sure your code works. It will crash in another place then 😄  #avoidtechincaldebt